source/develop/release-management/features/storage/multipath-events.html.md

```diff
@@ -0,0 +1,167 @@
+with the details of the event. Engine will create an audit log event. 
+Once the host is disconnected from the DC, Vdsm will unregister from
+the 'udev' events.
+
+
+### New vdsm API
+
+New multipath event type
+
+
+#### Multipath Event
+
+An event with information about the multipath device, path and status.
+
+A mapping with these keys:
+- Type: Path Failed/Path Reinstated (DM_ACTION)
+- Device (GUID): unique GUID of this device (DM_NAME)
+- TBD
+
+
+### Vdsm event monitoring
+
+A new framework in Vdsm is introduced to register to events:
+- A listener can register to a topic, providing a callback method/object.
+- A topic monitor will be created with the first listener.
+- Once the monitor receives an event, it will call the listener callback
+(in a different thread than the one the listener registered)
+- A listener can unregister from a topic.
+- The topic monitor will stop monitoring once the last listener unregister.
+
+#### Udev event monitor
+
+A Udev event monitor is introduced, listening to device-mapper udev events.
+The relevant events for this feature are PATH_FAILED and PATH_REINSTATED:
+
+Variable Name: DM_ACTION
+Uevent Action(s): KOBJ_CHANGE
+Type: string
+Description:
+Value: Device-mapper specific action that caused the uevent action.
+	PATH_FAILED - A path has failed.
+	PATH_REINSTATED - A path has been reinstated.
+
+- The Vdsm will register a listener to the Udev event monitor when connecting
+to the storage pool.
+- The listener will send an event to the engine when its callback is called.
+- The Vdsm will unregister the listener to the Udev event monitor when 
+disconnecting to the storage pool.
+- The user can disable this feature in a specific host in Vdsm configuration file
+
+### New engine API
+
+New audit logs are added:
+- Multipath device path up
+- Multipath device path down
+
+## Benefit to oVirt
+
+- The user will be able to detect storage issues faster.
+
+
+## User Experience
+
+New audit logs are added with information on the device path status.
+
+
+## Installation/Upgrade
+
+This feature does not affect engine or Vdsm installation or upgrades
+
+
+## User work-flows
+
+Once a user connects to an ISCSI server, 'Multipath device path up' event
+will be created.
+If there is an issue with the connection to the storage server, and the 
+multipath reports an issue with one or more device path,
+'Multipath device path down' will be created.
+Once the issue is fixed, 'Multipath device path up' event will be created.
+
+
+## Dependencies / Related Features
+
+Additional package dependency in Vdsm:
+- pyudev
+
+No related features.
+
+
+### Entity Description
+
+NA
+
+## CRUD
+
+NA
+
+
+## Event Reporting
+
+- 'Multipath device path up'
+- 'Multipath device path down'
+
+
+## Documentation / External references
+
+This feature is required for resolving
+[Bug 723931](https://bugzilla.redhat.com/723931)
+
+Pyudev - [Monitoring devices](http://pyudev.readthedocs.io/en/latest/guide.html#monitoring-devices)
```

Comment:
Please add also https://www.kernel.org/doc/Documentation/device-mapper/dm-uevent.txt

Reply:
it is here (below)